doUpsert/index.js

```diff
@@ -27,37 +27,100 @@ module.exports = async function(context, jsonItem) {
       new DefaultAzureCredential());
   const jsonString = JSON.stringify(jsonItem);
   context.log.verbose(`Json item: ${jsonString}`);
+  
   if ('$relationshipId' in jsonItem) {
-    context.log.verbose(`upserting relationship ${jsonItem.$relationshipId}`);
-    await (async () => {
-      context.log.verbose('waiting 100ms');
-      sleep(100);
-      context.log.verbose('calling ADT relationship API');
-      await digitalTwin.upsertRelationship(
-          jsonItem.$sourceId,
-          jsonItem.$relationshipId,
-          jsonItem.relationship)
-          .catch((e) => {
-            context.log.error(`relationship ${jsonItem.$relationshipId}
-              on source ${jsonItem.$sourceId} insertion failed: `, e);
-            const err = `failed relationship: ${jsonString}`;
-            throw err;
-          });
-    })();
+    if (jsonItem.relationship.$relationshipDelete === true) { // relationship must be deleted
+      context.log.verbose(`deleting relationship ${jsonItem.$relationshipId}`);
+      await (async () => {
+        context.log.verbose('waiting 100ms');
+        sleep(100);
+        context.log.verbose('calling ADT relationship API');
+        await digitalTwin.deleteRelationship(
+            jsonItem.$sourceId,
+            jsonItem.$relationshipId)
+            .catch((e) => {
+              context.log.error(`relationship ${jsonItem.$relationshipId}
+                on source ${jsonItem.$sourceId} deletion failed: `, e);
+              const err = `failed relationship: ${jsonString}`;
+              throw err;
+            });
+      })();
+    } else { // relationship must be upserted
+      context.log.verbose(`upserting relationship ${jsonItem.$relationshipId}`);
+      await (async () => {
+        context.log.verbose('waiting 100ms');
+        sleep(100);
+        context.log.verbose('calling ADT relationship API');
+        await digitalTwin.upsertRelationship(
+            jsonItem.$sourceId,
+            jsonItem.$relationshipId,
+            jsonItem.relationship)
+            .catch((e) => {
+              context.log.error(`relationship ${jsonItem.$relationshipId}
+                on source ${jsonItem.$sourceId} insertion failed: `, e);
+              const err = `failed relationship: ${jsonString}`;
+              throw err;
+            });
+      })();
+    }
   } else if ('$id' in jsonItem) {
-    context.log.verbose(`upserting twin ${jsonItem.$id}`);
-    // twin
-    await (async () => {
-      context.log.verbose('waiting 20ms');
-      sleep(20);
-      context.log.verbose('calling ADT twin API');
-      await digitalTwin.upsertDigitalTwin(jsonItem.$id, jsonString)
-          .catch((e) => {
-            context.log.error(`twin ${jsonItem.$id} insertion failed: `, e);
-            const err = `failed twin: ${jsonString}`;
-            throw err;
-          });
-    })();
+    if (jsonItem.$entityDelete === true) { // twin must be deleted
+      // list all twin relationships
+      let rels = []
```

Comment:
There is an inconsistency with usage of ";" at end of line. Can you please add them everywhere, this was started like that.
Je t'invite à passer `eslint doUpsert/index.js` sur le code, il y a d'autres erreurs:
/home/vcarluer/dev/phoenix/azure/azure-digital-twin-injector/doUpsert/index.js
  30:1   error  Trailing spaces not allowed                          no-trailing-spaces
  32:1   error  This line has a length of 93. Maximum allowed is 80  max-len
  69:11  error  'rels' is never reassigned. Use 'const' instead      prefer-const
  69:20  error  Missing semicolon                                    semi
  71:11  error  'out' is never reassigned. Use 'const' instead       prefer-const
  73:56  error  Missing semicolon                                    semi
  76:11  error  'inc' is never reassigned. Use 'const' instead       prefer-const
  78:54  error  Missing semicolon                                    semi